### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by vhlopezch**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Added my assignment with preprocessing, model training, hyperparameter tuning, evaluation, and export steps.

## What did you learn from the changes you have made?
How to build pipelines, tune models, evaluate performance, and export a trained model.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
I considered using SVR or Lasso, but stuck to Ridge and RandomForest for simplicity and performance.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Yes, SHAP failed so I need to learn more about it.

## How were these changes tested?
Via cross-validation and evaluation on a separate test set.

## A reference to a related issue in your repository (if applicable)
N/A
## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by Shiraz-Latif — APPROVED** [review]
Assignment-2 submission approved for partial scores.

Feedback: Shap for explainability is not completed.
